source/common/upstream/upstream_impl.cc

```diff
@@ -1208,6 +1208,11 @@ ClusterInfoImpl::ClusterInfoImpl(
                     envoy::config::cluster::v3::UpstreamConnectionOptions::HappyEyeballsConfig>(
                     config.upstream_connection_options().happy_eyeballs_config())
               : nullptr),
+      lrs_report_metric_names_(!config.lrs_report_endpoint_metrics().empty()
+                                   ? std::make_unique<Envoy::Orca::LrsReportMetricNames>(
+                                         config.lrs_report_endpoint_metrics().begin(),
+                                         config.lrs_report_endpoint_metrics().end())
+                                   : nullptr),
```

Comment:
And please also check the `lrs_report_endpoint_metrics` field of Cluster proto to see if there is a not-implemented-hide tag. If there is a related tag, please remove it.

And please ensure the API comments match to your implementation/behaviors.

Reply:
Done

Reply:
Oh, I haven't realized that this change going to be user-visible (if not, then I think the removal of the not-implemented-hide tag should be postponed).
If that's the case, there should be at least an integration-test that validates this feature.

Reply:
I think it would make sense to keep `[#not-implemented-hide:]` tag for now. 
We can add an integration test and remove the tag in the next PR.

Reply:
> And please also check the `lrs_report_endpoint_metrics` field of Cluster proto to see if there is a not-implemented-hide tag. If there is a related tag, please remove it.
> 
> And please ensure the API comments match to your implementation/behaviors.


@wbpcode Ito make the review PRs smaller (and more concise), I think enabling the API should be done in a followup PR that also includes an integration-test and a release note.
This way, if there's anything else that needs to be modified, we'll be able to review it more carefully.
WDYT?

Reply:
Okay, if you both think it's better. :)